nengo_viz/components/netgraph.py

```diff
@@ -21,19 +23,128 @@ def __init__(self, viz, config, uid):
         self.uids = {}
         self.parents = {}
         self.networks_to_search = [self.viz.model]
+        self.initialized_pan_and_zoom = False
+        self.last_modify_time = os.path.getmtime(self.viz.viz.filename)
 
-    def get_parents(self, uid):
+    def check_for_reload(self):
+        t = os.path.getmtime(self.viz.viz.filename)
+
+        if self.last_modify_time is None or self.last_modify_time < t:
+            self.reload()
+            self.last_modify_time = t
+
+    def reload(self):
+        locals = {}
+        with open(self.viz.viz.filename) as f:
+            code = f.read()
+        try:
+            exec(code, locals)
+        except:
+            traceback.print_exc()
+        try:
+            model = locals['model']
+        except:
+            traceback.print_exc()
+            return
+
+        locals['nengo_viz'] = nengo_viz
+        name_finder = nengo_viz.NameFinder(locals, model)
+
+        self.networks_to_search = [model]
+        self.parents = {}
+
+        for uid, old_item in self.uids.items():
```

Comment:
Might be a bit faster to use `iteritems` (or, for Python 3 compatibility, `nengo.utils.compat.iteritems`).